server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -941,35 +916,33 @@ private static Metadata metadataForSnapshot(SnapshotsInProgress.Entry snapshot,
         String repository,
         List<String> snapshots
     ) {
-        if (snapshotsInProgress == null || snapshotsInProgress.entries().isEmpty()) {
+        if (snapshotsInProgress == null || snapshotsInProgress.isEmpty()) {
             return Collections.emptyList();
         }
         if ("_all".equals(repository)) {
-            return snapshotsInProgress.entries();
+            return snapshotsInProgress.asStream().collect(Collectors.toUnmodifiableList());
         }
-        if (snapshotsInProgress.entries().size() == 1) {
+        if (snapshotsInProgress.count() == 1) {
```

Comment:
I think this all still works if we just look at the one repo, right?

```suggestion
        if (snapshotsInProgress.forRepo(repository).size() == 1) {
```

(also extract that variable sooner)

That said, not sure this special-case optimisation really adds much after these changes, I'd be fine with just dropping it.